### PR TITLE
Mothur

### PR DESCRIPTION
**Pull request by shiltemann**
Changed merge-files tool to use multiple=true instead of a repeat for the input files, and started a macro file (only has the requirement in it for now), and added some tests with handmade artificial data (maybe can replace later if somebody has good testing datasets)

and to make planemo happy I also: added citation (please double check), changed indentation, moved requirements to the top, changed format="input" to format_source

### CODE REVIEW

File: tools/mothur/tools/mothur/merge.files.xml [hunk 1]
```diff
@@ -1,64 +1,55 @@
...
+     --result='^mothur.\S+\.logfile$:'$logfile,'mergefile:'$output
+     --outputdir='$logfile.extra_files_path'
+     --input=${merge.inputs}
+     --output='mergefile'
+    ]]></command>
+    <inputs>
+    <conditional name="merge">
+        <param name="type" type="select" format="fasta,qual,groups,names,accnos" label="Merge">
+            <option value="fasta">fasta</option>
+            <option value="qual">qual</option>
```
Comment (bgruening) [inline tools/mothur/tools/mothur/merge.files.xml]: @shiltemann I guess we can replace this entirely by:

<param name="inputs" type="data" format="fasta,qual,groups,names,accnos" multiple="true" label="inputs - fasta"/>

Do we need the select box for datatype?

File: tools/mothur/tools/mothur/merge.files.xml [hunk 2]
```diff
@@ -74,5 +65,18 @@ The merge.files_ command merge inputs into a single output.
...
 
- </help>
+    </help>
+    <citations>
+        <citation type="bibtex">
+@article{mothur,
+  author = {Schloss PD, Westcott SL, Ryabin R, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF},
+  year = {2009},
+  volume = {75},
+  number = {23},
```
Comment (bgruening) [inline tools/mothur/tools/mothur/merge.files.xml]: here I would use the doi:

<citation type="doi">10.1128/AEM.01541-09</citation>

**Comment by bgruening** [comment]
Otherwise, this looks great!

File: tools/mothur/tools/mothur/merge.files.xml (hunk 1 above)
Comment (shiltemann) [inline tools/mothur/tools/mothur/merge.files.xml]: that would be fine by me, I just don't know how much users want/need/like the filtering on their inputs to ensure they select all files of the same and intended datatype.. maybe some of the others can weigh in on this too..

Comment (erasche) [inline tools/mothur/tools/mothur/merge.files.xml]: If all files must be of the same type, then I'm in favour of @shiltemann's approach here. If mixed are a allowed I lean towards @bgruening's.

Comment (jj-umn) [inline tools/mothur/tools/mothur/merge.files.xml]: I think this only makes sense if the files are all of the same type.

Comment (shiltemann) [inline tools/mothur/tools/mothur/merge.files.xml]: As far as I understand it (but I'm no mothur expert), you would only ever do a merge on files of the same data type. If we can add a validator to the param to check all files are same format that would probably work too, but I don't know if that's possible?

Comment (erasche) [inline tools/mothur/tools/mothur/merge.files.xml]: This is probably easier to write than a validator and clearer to the users about the expectations.

**Pull request merged**